### PR TITLE
Smrt enrico consolidation

### PR DESCRIPTION
**Pull request by aprilnovak**
This PR has a couple first steps in eventually having `ShiftNekDriver` derive from `CoupledDriver`. This PR adds:

1. Mostly empty `SmrtCoupledDriver` class to assist in incremental updates to `ShiftNekDriver` so that we will eventually be able to replace `SmrtCoupledDriver` by `CoupledDriver`.

2. Split out some actions from the `solve` methods of `smrt` classes into more specific, smaller tasks.

3. Moved the definition of the position MPI datatype to `message_passing.h` since it's used equivalently by `ShiftNekDriver` and `OpenmcNekDriver`.

@sphamil would you please verify that this compiles? Hopefully I didn't leave any typos...

### CODE REVIEW

**Comment by sphamil** [comment]
A couple minor issues, here's the diff:
```
->git diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1fad62e..ffe4e4b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,6 +84,7 @@ if (SCALE_FOUND)
         src/smrt/Multi_Pin_Conduction.cpp
         src/smrt/Multi_Pin_Subchannel.cpp
         src/smrt/shift_driver.cpp
+        src/smrt/smrt_coupled_driver.cpp
         src/smrt/Single_Pin_Conduction.cpp
         src/smrt/Single_Pin_Subchannel.cpp
         src/smrt/Two_Group_Cross_Sections.cpp
diff --git a/src/smrt/shift_nek_driver.cpp b/src/smrt/shift_nek_driver.cpp
index 95f4591..2e522ac 100644
--- a/src/smrt/shift_nek_driver.cpp
+++ b/src/smrt/shift_nek_driver.cpp
@@ -160,7 +160,7 @@ void ShiftNekDriver::normalize_power()
 // Currently, this sets up only position_mpi_datatype
 void ShiftNekDriver::init_mpi_datatypes()
 {
-  d_position_mpi_type = define_position_mpi_type();
+  d_position_mpi_type = define_position_mpi_datatype();
 }

 // Free user-defined MPI types
diff --git a/src/smrt/smrt_coupled_driver.cpp b/src/smrt/smrt_coupled_driver.cpp
index 72d34cf..53846bb 100644
--- a/src/smrt/smrt_coupled_driver.cpp
+++ b/src/smrt/smrt_coupled_driver.cpp
@@ -1,3 +1,5 @@
+#include <stdexcept>
+#include "pugixml.hpp"
 #include "smrt/smrt_coupled_driver.h"

 namespace enrico {
```

File: src/smrt/smrt_coupled_driver.cpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#include "smrt/smrt_coupled_driver.h"
+#include "pugixml.hpp""
```
Comment (paulromano) [inline src/smrt/smrt_coupled_driver.cpp]: Extra quote character here

Comment (aprilnovak) [reply src/smrt/smrt_coupled_driver.cpp]: Ah, good catch!

**Comment by paulromano** [comment]
Also have a merge conflict on this PR

**Comment by aprilnovak** [comment]
Thanks for pointing those two out, I'll get to this this evening

File: include/enrico/openmc_nek_driver.h [hunk 2]
```diff
@@ -47,9 +48,9 @@ class OpenmcNekDriver : public CoupledDriver {
   bool is_converged() override;
 
   Comm intranode_comm_; //!< The communicator representing intranode ranks
-  std::unique_ptr<OpenmcDriver> openmc_driver_; //!< The OpenMC driver
-  std::unique_ptr<NekDriver> nek_driver_;       //!< The Nek5000 driver
-  int openmc_procs_per_node_; //!< Number of MPI ranks per (shared-memory) node in OpenMC comm
+  double pressure_;                             //!< System pressure in [MPa]
```
Comment (paulromano) [inline include/enrico/openmc_nek_driver.h]: Looks like the `pressure_` member crept back in during your rebase. Can you remove this?

File: src/message_passing.cpp [hunk 3]
```diff
@@ -30,4 +30,30 @@ void get_node_comms(MPI_Comm super_comm,
     MPI_Comm_free(sub_comm);
 }
 
+MPI_Datatype define_position_mpi_datatype()
+{
+  MPI_Datatype type;
```
Comment (paulromano) [inline src/message_passing.cpp]: I would put this line down immediately before the call to `MPI_Type_create_struct`

**Comment by paulromano** [comment]
Thanks @aprilnovak!

**Pull request merged**